### PR TITLE
Fixed grammatical errors in the first line.

### PR DESCRIPTION
**Pull request by Swapnil-11-04-01**
Hi there,

Upon reviewing the `readme`, I identified a few minor grammatical errors and have suggested appropriate revisions to enhance the professional appeal of the document.

1.  ```diff
    - This repository is provides
    + This repository provides
    ```
1.  ```diff
    - tooling needed to compile
    + necessary tooling to compile
    ```
1.  ```diff
    - produce a HEX file
    + generates a HEX file
    ```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 [![Native Build Status](https://github.com/lancaster-university/microbit-v2-samples/actions/workflows/build.yml/badge.svg)](https://github.com/lancaster-university/microbit-v2-samples/actions/workflows/build.yml) [![Docker Build Status](https://github.com/lancaster-university/microbit-v2-samples/actions/workflows/docker-image.yml/badge.svg)](https://github.com/lancaster-university/microbit-v2-samples/actions/workflows/docker-image.yml)
 
-This repository is provides the tooling needed to compile a C/C++ CODAL program for the micro:bit v2 and produce a HEX file that can be downloaded to the device.
+This repository provides the necessary tooling to compile a C/C++ CODAL program for the micro:bit v2 and generates a HEX file that can be downloaded to the device.
```
Comment (microbit-carlos) [inline README.md:5]: Thanks for the Pr @Swapnil-11-04-01!

For the last word `generates`, I believe it should be `generate` as it is meant to read "to compile ... and generate" not "to compile .. and generates".

Comment (microbit-carlos) [reply README.md:5]: I'll apply this patch and merge, thanks @Swapnil-11-04-01! 
```suggestion
This repository provides the necessary tooling to compile a C/C++ CODAL program for the micro:bit v2 and generate a HEX file that can be downloaded to the device.
```

**Review by microbit-carlos — APPROVED** [review]
Thanks again for the PR @Swapnil-11-04-01!

File: README.md (hunk 1 above)
Comment (Swapnil-11-04-01) [reply README.md]: Thanks, `@microbit-carlos`  for clarifying and patching the above. Also, I apologize for the late response, I am new in Open Source Contributions, therefore still learning and getting used to it.

Comment (microbit-carlos) [reply README.md]: No worries, your contribution is appreciated, thanks for the PR!

Comment (Swapnil-11-04-01) [reply README.md]: Thanks!

**Pull request merged**